MonoGame.Framework/Graphics/GraphicsDevice.cs

```diff
@@ -159,6 +159,20 @@ public GraphicsAdapter Adapter
             private set;
         }
 
+        internal Metrics _metrics;
+
+        /// <summary>
+        /// The rendering data for debugging and profiling. 
+        /// </summary>
+        public Metrics Metrics { get { return _metrics; } }

```

Comment:
No reason why we couldn't allow someone to set a new set of metrics.  This allows you to store some metrics, reset and do some other work, and restore them later to continue.